module/builder/collection/builder_test.go

```diff
@@ -77,10 +79,22 @@ func (suite *BuilderSuite) SetupTest() {
 	suite.Require().Nil(err)
 	suite.mutator = suite.state.Mutate()
```

Comment:
Why do we still need `Mutate()` function here?

Reply:
it's a different module, I was refactoring `protocol.State`, this is `cluster.state`

Reply:
Yup, so the question is what's plan moving forward with the cluster state? Before this refactor, both cluster state and protocol state have the same design which uses `Mutate()` to return a Mutable state for mutation. Now the protocol state has changed, but the cluster state still kept the `Mutate()` method. To me, ideally they could follow the same design, because otherwise people have to learn two different ways to mutate two different states.

Reply:
Yeah, I see your point and I agree with it, the only thing that I am not sure if we should spend time on this right now or postpone it for later since it cluster mutator is not a blocker for anything at the moment

Reply:
Yeah, not right now. We could work on it in the future.